test/e2e/framework/util.go

```diff
@@ -1536,7 +1536,7 @@ func headersForConfig(c *restclient.Config, url *url.URL) (http.Header, error) {
 	if err != nil {
 		return nil, err
 	}
-	if _, err := rt.RoundTrip(&http.Request{URL: url}); err != nil {
+	if _, err := rt.RoundTrip(&http.Request{Header: make(http.Header), URL: url}); err != nil {
```

Comment:
Why doesn't this use [http.NewRequest](https://golang.org/pkg/net/http/#NewRequest)? are there other things wrong with this construction?

Reply:
I don't know, I guess the original author avoided a constructor to show that this is a "dummy" request whose only purpose is to figure out Headers.

Either way, I did the bare minimum change to fix my problem, but best practice would probably be to do

`NewRequest("GET", url, nil)`.

So let me open another PR.